src/patchelf.cc

```diff
@@ -1237,7 +1237,7 @@ template<ElfFileParams>
 std::string ElfFile<ElfFileParamNames>::getInterpreter()
 {
     auto shdr = findSection(".interp");
-    return std::string((char *) fileContents->data() + rdi(shdr.sh_offset), rdi(shdr.sh_size));
+    return std::string((char *) fileContents->data() + rdi(shdr.sh_offset), rdi(shdr.sh_size) - 1);
```

Comment:
I am trying to understand why the additional 1 here was an additional null character.

Reply:
Ah okay, when you construct a `std::string` from a `char *` you *should not* take into account the `\0` in the length.